client/layout/guided-tours/config-elements/step.tsx

```diff
@@ -168,6 +168,15 @@ export default class Step extends Component< Props, State > {
 			}
 		}
 
+		if ( props.target ) {
+			return new Promise( resolve => {
```

Comment:
I think we should only return this `Promise` if `props.target` has changed compared to its previous value...

Reply:
This sort of check will become much easier once #33409 is merged, since it implements a similar concept for `props.wait`.